polaris-migrator/src/migrations/replace-sass-lengths/replace-sass-lengths.ts

```diff
@@ -0,0 +1,187 @@
+import type {FileInfo, API, Options} from 'jscodeshift';
+import postcss, {Plugin} from 'postcss';
+import valueParser, {Node} from 'postcss-value-parser';
+import {toPx} from '@shopify/polaris-tokens';
+
+import {POLARIS_MIGRATOR_COMMENT} from '../../constants';
+import {
+  NamespaceOptions,
+  namespace,
+  createIsSassFunction,
+} from '../../utilities/sass';
+
+// List of the props we want to run this migration on
+const targetProps = [
+  'padding',
+  'padding-top',
+  'padding-right',
+  'padding-bottom',
+  'padding-left',
+  'padding-inline',
+  'padding-inline-start',
+  'padding-inline-end',
+  'padding-block',
+  'padding-block-start',
+  'padding-block-end',
+  'margin',
+  'margin-top',
+  'margin-right',
+  'margin-bottom',
+  'margin-left',
+  'margin-inline',
+  'margin-inline-start',
+  'margin-inline-end',
+  'margin-block',
+  'margin-block-start',
+  'margin-block-end',
+] as const;
+
+type TargetProp = typeof targetProps[number];
+
+const isTargetProp = (propName: unknown): propName is TargetProp =>
+  targetProps.includes(propName as TargetProp);
+
+// Mapping of spacing tokens and their corresponding px values
+const spacingTokensMap = {
+  '0': '--p-space-0',
+  '0px': '--p-space-0',
```

Comment:
Not directly related to this migration, but more around our tokens... **Do we need or should we have a token in a scaled system for values of `0`?**

From my understanding, this wouldn't be something we'd want adjust and it don't provide much in terms of "guidance" of allowable values.

Maybe a hot take, but `0` should always be allowable for styling. If so, then we would need to update our coverage rules. Perhaps I'm missing a use-case, but the `--p-space-0` token could be unnecessary and only adding to more complexity (i.e. adding tokens that don't help serve the system or improve crafter experience).

```css
/* current practice recommendation (always use tokens) */
padding: var(--p-space-0) var(--p-space-4);

/* new recommendation (use tokens were they add guidance */
padding: 0 var(--p-space-4);
```

In context of this migration, I would suggest we disregard usages of `0` and `0px` with the plan to then:

- update `0` as a valid coverage value
- and deprecate/remove the `--p-space-0` token

Thoughts @lgriffee @aaronccasanova ?

Reply:
I'm actually a fan of `0`s for token scales. In my experience they are nice for two things:

1. Clean/programmatic logic e.g.
```js
const styles = { boxShadow: `var(--p-shadows-${elevated ? 2 : 0})` }
```

2. Pivot point for inverse tokens where the names of negative tokens are symmetric to their `> 0` counter part:
```css
--space--3: calc(-3 * var(--p-space-3));
--space--2: calc(-2 * var(--p-space-2));
--space--1: calc(-1 * var(--p-space-1));
--space-0: 0px;
--space-1: 2px;
--space-2: 4px;
--space-3: 6px;
```

That said, we still need to have more discussions around token scale naming/conventions. I'm fine with updating the script to skip `0` lengths for now, but don't think we should remove the `0` token from our scale.

Reply:
Good points 👌 Certainly don't have to address this topic here & now, but thought I'd inquire and challenge the need.

I may move this to a discussion, but for now I'd say lets keep the migration as-is and address `0`'s to `var(--p-space-0)`.